test/rage2e/rag_test.go

```diff
@@ -0,0 +1,340 @@
+				return nil
+			}
+			if err != nil {
+				return fmt.Errorf("error checking if ragengine %s exists: %v", ragengineObj.Name, err)
+			}
+
+			err = utils.TestingCluster.KubeClient.Delete(ctx, ragengineObj, &client.DeleteOptions{})
+			if err != nil {
+				return fmt.Errorf("failed to delete ragengine %s: %v", ragengineObj.Name, err)
+			}
+			return nil
+		}, utils.PollTimeout, utils.PollInterval).Should(Succeed(), "Failed to delete ragengine")
+	})
+
+	return nil
+}
+
+// validateInferenceResource validates inference deployment
+func validateInferenceandRAGResource(objectMeta metav1.ObjectMeta, expectedReplicas int32, isStatefulSet bool) {
+	By("Checking the inference resource", func() {
+		Eventually(func() bool {
+			var err error
+			var readyReplicas int32
+
+			if isStatefulSet {
+				sts := &appsv1.StatefulSet{
+					ObjectMeta: metav1.ObjectMeta{
+						Name:      objectMeta.Name,
+						Namespace: objectMeta.Namespace,
+					},
+				}
+				err = utils.TestingCluster.KubeClient.Get(ctx, client.ObjectKey{
+					Namespace: objectMeta.Namespace,
+					Name:      objectMeta.Name,
+				}, sts)
+				readyReplicas = sts.Status.ReadyReplicas
+
+			} else {
+				dep := &appsv1.Deployment{
+					ObjectMeta: metav1.ObjectMeta{
+						Name:      objectMeta.Name,
+						Namespace: objectMeta.Namespace,
+					},
+				}
+				err = utils.TestingCluster.KubeClient.Get(ctx, client.ObjectKey{
+					Namespace: objectMeta.Namespace,
+					Name:      objectMeta.Name,
+				}, dep)
+				readyReplicas = dep.Status.ReadyReplicas
+			}
+
+			if err != nil {
+				GinkgoWriter.Printf("Error fetching resource: %v\n", err)
+				return false
+			}
+
+			if readyReplicas == expectedReplicas {
+				return true
+			}
+
+			return false
+		}, 20*time.Minute, utils.PollInterval).Should(BeTrue(), "Failed to wait for inference resource to be ready")
+	})
+}
+
+func validateAssociatedService(objectMeta metav1.ObjectMeta) {
+	serviceName := objectMeta.Name
+	serviceNamespace := objectMeta.Namespace
+
+	By(fmt.Sprintf("Checking for service %s in namespace %s", serviceName, serviceNamespace), func() {
+		service := &v1.Service{}
+
+		Eventually(func() bool {
+			err := utils.TestingCluster.KubeClient.Get(ctx, client.ObjectKey{
+				Namespace: serviceNamespace,
+				Name:      serviceName,
+			}, service)
+
+			if err != nil {
+				if errors.IsNotFound(err) {
+					GinkgoWriter.Printf("Service %s not found in namespace %s\n", serviceName, serviceNamespace)
+				} else {
+					GinkgoWriter.Printf("Error fetching service %s in namespace %s: %v\n", serviceName, serviceNamespace, err)
+				}
+				return false
+			}
+
+			GinkgoWriter.Printf("Found service: %s in namespace %s\n", serviceName, serviceNamespace)
+			return true
+		}, 10*time.Minute, utils.PollInterval).Should(BeTrue(), "Failed to wait for service to be created")
+	})
+}
+
+// validateRAGEngineReadiness validates ragengine readiness
+func validateRAGEngineReadiness(ragengineObj *kaitov1alpha1.RAGEngine) {
```

Comment:
Why do you have both validateRAGEngineReadiness and validateRAGEngineResourceStatus? can they be merged?